torch/nn/_functions/thnn/activation.py

```diff
@@ -126,6 +127,56 @@ def backward(self, grad_output):
         return grad_input.mul(-1)
 
 
+class Threshold(Function):
+
+    @staticmethod
+    def forward(ctx, input, threshold, value, inplace):
+        if inplace:
+            if value > threshold:
+                raise RuntimeError('in-place processing requires value ({}) to not '
+                                   'exceed threshold ({})'.format(value, threshold))
+        ctx.threshold = threshold
+        ctx.value = value
+        ctx.inplace = inplace
+
+        if inplace:
+            ctx.mark_dirty(input)
+        output = input.new(input.size())
+        ctx.save_for_backward(input)
+
+        backend = type2backend[type(input)]
+        backend.Threshold_updateOutput(
+            backend.library_state,
+            input,
+            output,
+            threshold,
+            value,
+            inplace
+        )
+        return output
+
+    @staticmethod
+    def backward(ctx, grad_output):
+        input, = ctx.saved_variables
+        if grad_output.volatile:
+            grad_input = Variable(input.data.new(input.size()), volatile=True)
+            backend = type2backend[type(input.data)]
+            backend.Threshold_updateGradInput(
+                backend.library_state,
+                input.data,
+                grad_output.data,
+                grad_input.data,
+                ctx.threshold,
+                ctx.value,
+                False
+            )
+        else:
+            mask = input > ctx.threshold
+            grad_input = mask.type_as(grad_output) * grad_output
```

Comment:
I think `grad_input = grad_output.masked_fill(input > ctx.threshold, 0)` would be more efficient.